packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,146 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/foundation.dart';
+
+import 'package:flutter/cupertino.dart';
+import 'shared_state.dart';
+
+class SegmentedControl extends StatefulWidget {
+  final List<Widget> options;
+
+  const SegmentedControl({Key key, @required this.options}) : super(key: key);
+
+  @override
+  _SegmentedControlState createState() => new _SegmentedControlState();
+}
+
+class _SegmentedControlState extends State<SegmentedControl> {
+  int sharedState = 0;
+
+  @override
+  Widget build(BuildContext context) {
+    List<Widget> buttons = new List<Widget>();
+
+    int index = 0;
+    for (int i = 0; i < widget.options.length; i += 1, index += 2) {
+      buttons.add(new Expanded(child: new SegmentedControlButton(widget.options.elementAt(i), index)));
+      if (i != widget.options.length - 1) {
+        buttons.add(new SegmentedControlButton(
+            new Container(
+                decoration:
+                new BoxDecoration(color: CupertinoColors.activeBlue),
+                constraints:
+                new BoxConstraints(maxWidth: 2.0, minWidth: 2.0)),
+            index + 1));
+      }
+    }
+
+    return Container(
+        alignment: Alignment.center,
+        color: const Color(0xFFFFFFFF),
+        child: new Padding(
+            padding: EdgeInsets.symmetric(horizontal: 5.0),
```

Comment:
Any "magic numbers" like the 5.0 here should be `static const` constants that have names that describe their use: here it could be called `horizontalItemPadding`, for instance.